### ISSUE TITLE
AC-3 5.1 downgrade to AC-3 stereo

### ISSUE THREAD

**Issue by changxiangzhong** [gpac/gpac#1624]
- [X] I looked for a similar issue and couldn't find any.
- [X] I tried with the latest version of GPAC. Installers available at http://gpac.io/downloads/gpac-nightly-builds/
- [X] I give enough information for contributors to reproduce my issue (meaningful title, github labels, platform and compiler, command-line ...). I can share files anonymously with this dropbox: https://www.mediafire.com/filedrop/filedrop_hosted.php?drop=eec9e058a9486fe4e99c33021481d9e1826ca9dbc242a6cfaab0fe95da5e5d95

I've used the following steps to convert the [Joker trailer](https://download.wetransfer.com//us2/08b347559143454a00c75befbdd0fed120190813110311/de3194c367cb64290ebffec5d680e90e050eb974/JOKER_TSR-1_1920X1080_AC3_51_UP-thedigitaltheater.mkv?cf=y&token=eyJhbGciOiJIUzI1NiJ9.eyJleHAiOjE2MDI3NDk3NDQsInVuaXF1ZSI6IjA4YjM0NzU1OTE0MzQ1NGEwMGM3NWJlZmJkZDBmZWQxMjAxOTA4MTMxMTAzMTEiLCJmaWxlbmFtZSI6IkpPS0VSX1RTUi0xXzE5MjBYMTA4MF9BQzNfNTFfVVAtdGhlZGlnaXRhbHRoZWF0ZXIubWt2Iiwid2F5YmlsbF91cmwiOiJodHRwOi8vcHJvZHVjdGlvbi5iYWNrZW5kLnNlcnZpY2UudXMtZWFzdC0xLnd0OjkyOTIvd2F5YmlsbC92MS9zYXJrYXIvNjZiMmEwNmM2ODU1NGU1NzkxNGVlNjZhMjIxYzQ3MzY1NmU2NjlkMThlZWY4NmZjZGI3NzU0MGYwZGJhNTM4MTZjZjM0OTAwODA0MTdmZjI4ZjJlZmIiLCJmaW5nZXJwcmludCI6ImRlMzE5NGMzNjdjYjY0MjkwZWJmZmVjNWQ2ODBlOTBlMDUwZWI5NzQiLCJjYWxsYmFjayI6IntcImZvcm1kYXRhXCI6e1wiYWN0aW9uXCI6XCJodHRwOi8vcHJvZHVjdGlvbi5mcm9udGVuZC5zZXJ2aWNlLmV1LXdlc3QtMS53dDozMDAwL3dlYmhvb2tzL2JhY2tlbmRcIn0sXCJmb3JtXCI6e1widHJhbnNmZXJfaWRcIjpcIjA4YjM0NzU1OTE0MzQ1NGEwMGM3NWJlZmJkZDBmZWQxMjAxOTA4MTMxMTAzMTFcIixcImRvd25sb2FkX2lkXCI6MTA0ODA3MzQwMDZ9fSJ9.uIGE98K9ox0ebgLzxBDBFuPiRM1BkqKO-LN4RZ_yNB4) into the dash format. **But surprisingly, the AC-3 5.1 degraded to AC-3 Stereo**. Really appreciate if someone can explain to me how to keep the 5.1 channels.

Steps I've used.

- Download the joker trailer
- Change the container file mkv->mp4 by `ffmpeg -i joker.mkv -codec copy joker.mp4`
- Examine the audio format and get the result as following
```
hang@p300a:~/git-root/leixiaohua-ffmpeg/CDxFFmpeg$ ffprobe joker.mp4 
ffprobe version 4.2.4 Copyright (c) 2007-2020 the FFmpeg developers
   Stream #0:1(eng): Audio: ac3 (ac-3 / 0x332D6361), 48000 Hz, 5.1(side), fltp, 640 kb/s (default)
```
 - Execute the following to get dash file chunks
```
 MP4Box -dash 2000 -rap -frag-rap -url-template -dash-profile live -init-seg '$RepresentationId$/init.mp4' -out playlist.mpd joker.mp4
```
 - Deploy all the output files into apache2 server
 - Examine dash stream. And surprisingly, it's ac-3 stereo now.
```
chang@p300a:~/git-root/leixiaohua-ffmpeg/CDxFFmpeg$ ffprobe http://cxz.mynetgear.com/dash/joker/playlist.mpd
    Stream #0:1: Audio: ac3 (ac-3 / 0x332D6361), 48000 Hz, stereo, fltp, 632 kb/s
    Metadata:
      variant_bitrate : 639898
      id              : 1

```

I'm using the latest MP4Box 1.0.1.

**Comment by jeanlf** [comment]
I can't download your test sample, could you upload it somewhere else?

**Comment by changxiangzhong** [comment]
Wow! God speed! yeah, sure, I'll see what I can do.

**Comment by jeanlf** [comment]
indeed we did not support dolby specific signaling. Now fixed, aligned with DASH IOP v4.3. Thanks for the report !

**Comment by changxiangzhong** [comment]
Regarding my last step to convert MP4 file into Dash chunks. The script is wrong. I have to apply the following scripts to achieve that

```
 2061  ffmpeg -i ../joker.mkv -c copy -vn joker_audio.mp4
 2063  ffmpeg -i ../joker.mkv -c copy -an joker_video.mp4
 2068  MP4Box -dash 2000 -rap -frag-rap -url-template -dash-profile live -init-seg '$RepresentationId$/init.mp4' -out playlist.mpd joker_audio.mp4  joker_video.mp4 
 2069  ffplay http://localhost/dash/joker/playlist.mpd
```
